### PR TITLE
Fix: ItemList selection bug && navigation bug

### LINKED ISSUE #37277: ItemList disabled items trigger selection and activation signals on mouse
**Godot version:**
3.2.1.stable

**OS/device including version:**
Windows 10

**Issue description:**
According to `ItemList` documentation setting an item disabled means:
> Disabled items cannot be selected and do not trigger activation signals (when double-clicking or pressing Enter).

However, when clicked with mouse, they can still trigger `item_selected` and `item_activated` signals. Visually such items appear disabled and they cannot be selected with keyboard either: pressing arrow keys does not move the selection if the next item is disabled. But on clicks they do react.

Selection itself never changes, and signals can be filtered with `is_item_disabled`. So, this may be an error in classref. The only similar issue I have found was related to 2.1 (#5683), which resulted in fixing item behaviour.

**Steps to reproduce:**
1. Create a UI scene with an `ItemList` node.
2. Create a similar script to fill `ItemList` with items and disable one:
```swift
extends Control

onready var item_list : ItemList = $ItemList

func _ready() -> void:
	item_list.clear()
	item_list.add_item("test item 1")
	item_list.add_item("test item 2")
	item_list.add_item("test item 3")
	item_list.add_item("test item 4")
	item_list.add_item("test item 5")
	
	item_list.set_item_disabled(2, true)
	
	item_list.connect("item_activated", self, "_on_item_activated")
	item_list.connect("item_selected", self, "_on_item_selected")

func _on_item_activated(item_index : int) -> void:
	print("item #" + str(item_index) + " was activated")

func _on_item_selected(item_index : int) -> void:
	print("item #" + str(item_index) + " was selected")
```
3. Run the project, clicking on the disabled item still produces prints:
>item #2 was selected
item #2 was activated

**Minimal reproduction project:**
[bug-itemlist-disabled-activation.zip](https://github.com/godotengine/godot/files/4375668/bug-itemlist-disabled-activation.zip)

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #37277
Fix: ItemList navigation with disabled items

list-item-nav

As far I know `items[i].selectable` and `!items[i].disabled` are interchangeable here. I think we should use disabled only for rendering stuff and selected for selection. 
https://github.com/godotengine/godot/blob/e9ecb06cf36fbf1f9174a3c4727416194d60d7aa/scene/gui/item_list.cpp#L266

### CODE REVIEW

File: scene/gui/item_list.cpp [hunk 1]
```diff
@@ -32,6 +32,9 @@
 #include "core/os/os.h"
 #include "core/project_settings.h"
 
+#define CAN_SELECT(i) ((items[i].selectable) && (!items[i].disabled))
+#define IS_SAME_ROW(i, row) (i / current_columns == row)
```
Comment (akien-mga) [inline scene/gui/item_list.cpp]: Please `#undef` them after the method.
I'd suggest also moving the `#define` directly in the relevant code branch of `_gui_input` where they'll be used (and undefine at the end of that branch).

**Comment by pycbouh** [comment]
> I think we should use disabled only for rendering stuff and selected for selection.

If that is the solution, then please update the class reference, so that it does not suggest otherwise. The issue you are linking is about signals getting triggered in cases that the docs suggest they shouldn't.

**Comment by ThakeeNathees** [comment]
> Is that correct? If so, this PR does not fix #37277.

In my fix it check whether the item can be selected or not, by it's not disabled and selectable (the property of an item) and it solves #37277 

but I think we don't have to check if it's disabled or not for selection.

@pycbouh

**Comment by groud** [comment]
Sorry for the late review. The fix looks good but the PR needs a rebase.

**Comment by KoBeWi** [comment]
Superseded by #60123

**Pull request closed without merging**